modules/core/src/encoder/field/PurposeRestrictionVectorEncoder.ts

```diff
@@ -44,8 +44,11 @@ export class PurposeRestrictionVectorEncoder {
 
           }
 
-          // either end of the loop or there's a gap greater than 1 number
-          if (i === len - 1 || vendors[i + 1] > vendorId + 1) {
+          /**
+           * either end of the loop or there's a gap greater than 1 number and
+           * the vendorId is in the GVL
+           */
+          if (i === len - 1 || (vendors[i + 1] > vendorId + 1 && prVector.gvl.vendorIds.has(vendorId + 1))) {
```

Comment:
this isn't quite right, as it basically ignores gaps unless the gvl has `vendorId+1`. The logic needs to be "the next vendor in the restrictions list (`vendors[i+1]`) is greater than the next vendor in the gvl".

Reply:
This is a bit tricky to do in linear-time on a javascript Set, but I think something like this will work and be efficient-enough:
```
const lastVendorId = vendors[len-1]; // we know len>0 since we're in the loop
const gvlVendorIds = prVector.gvl.vendorIds;
const nextGvlVendor = function(vendorId) {
  while(++vendorId <= endVendorId && !gvlVendorIds.has(vendorId) {
  }
  return vendorId;
}

if(i === len-1 || vendors[i+1] > nextGvlVendor(vendorId)) {
```

Reply:
(if we wanted to make that a bit more efficient, could write it as `nextVendorIsAfterNextGvlVendor(vendorId, vendors[i+1])` so we could stop the loop as soon as we know we fail).

Reply:
Thank you @a2intl for your feedback.

The PR has been updated and contains the logic as suggested above.